standard/expressions.md

```diff
@@ -3219,14 +3206,12 @@ When a *stackalloc_initializer* is present:
 
 Each *stackalloc_element_initializer* shall have an implicit conversion to *unmanaged_type* ([§10.2](conversions.md#102-implicit-conversions)). The *stackalloc_element_initializer*s initialize elements in the allocated memory in increasing order, starting with the element at index zero. In the absence of a *stackalloc_initializer*, the content of the newly allocated memory is undefined.
 
-The result of a *stackalloc_expression* is an instance of type `Span<T>`, where `T` is the *unmanaged_type*:
+If a *stackalloc_expression* occurs directly as the initializing expression of a *local_variable_declaration* ([§13.6.2](statements.md#1362-local-variable-declarations)), where the *local_variable_type* is either a pointer type ([§23.3](unsafe-code.md#233-pointer-types)) or inferred (`var`), then the result of the *stackalloc_expression* is a pointer of type `T *` ([§23.9](????)). In this case the *stackalloc_expression* must appear in unsafe code. Otherwise the result of a *stackalloc_expression* is an instance of type `Span<T>`, where `T` is the *unmanaged_type*:
```

Comment:
This appears to be a breaking change from v7 – where `stackalloc` is defined to be of type `Span<T>` in a safe context and there is no restriction on using it as the initialiser in an inferred (`var`) declaration.

Reply:
The type is inferred to be a pointer here, because the compiler preserves backward compatibility:
```cs
var pointer = stackalloc int[1];
```
And a span here:
```cs
var span = (stackalloc int[1]);
```

Reply:
And, in a safe context, `var x = stackalloc` is still not permitted, because it does continue inferring a pointer type.